addons/hr_recruitment_integration_base/models/hr_job_post.py

```diff
@@ -0,0 +1,204 @@
+                author_id=self.env.user.partner_id.id,
+                subject=_(
+                    'Job Post on %(platform)s has been modified',
+                    platform=post.platform.name
+                ),
+                body=Markup
+                    (_(
+                        '''
+                        <strong>Job Post on %(platform)s has been %(mode)s</strong> <br/>
+                        <strong>Job:</strong>  %(job)s\n<br/>
+                        <strong>Status:</strong>  %(status)s<br/>
+                        <strong>Status Message:</strong> %(status_message)s<br/>
+                        <strong>Contact Method:</strong> %(method)s<br/>
+                        <strong>Contact Point:</strong> %(vector)s<br/>
+                        <strong>Campaign Start Date:</strong> %(start_date)s<br/>
+                        <strong>Campaign End Date:</strong> %(end_date)s <br/>
+                        <strong>Company:</strong> %(company)s
+                        ''',
+                        platform=post.platform.name,
+                        mode=mode,
+                        job=post.job_id.name,
+                        status=dict(post._fields['status'].selection).get(post.status),
+                        status_message=post.status_message,
+                        method=dict(post._fields['apply_method'].selection).get(post.apply_method),
+                        vector=post.apply_vector,
+                        start_date=post.campaign_start_date,
+                        end_date=post.campaign_end_date or _('No Limit'),
+                        company=post.company_id.name,
+                    )
+                ),
+            )
+
+    def _end_finished_campaign(self):
+        posts_to_end = self.env['hr.job.post'].search([
+            ('status', 'not in', ['deleted', 'expired']),
+            ('campaign_end_date', '<=', fields.Date.today()),
+        ])
+        for post in posts_to_end:
+            post.action_delete_job_post()
+
+    def _start_new_campaign(self):
+        posts_to_start = self.env['hr.job.post'].search([
+            ('status', '=', 'pending'),
+            ('campaign_start_date', '<=', fields.Date.today()),
+            '|',
+                ('campaign_end_date', '>=', fields.Date.today()),
+                ('campaign_end_date', '=', False),
+        ])
+        for post in posts_to_start:
+            wizard = self.env['hr.recruitment.post.job.wizard'].create({
+                'job_id': post.job_id.id,
+                'apply_method': post.apply_method,
+                'platforms': [(6, 0, post.platform.ids)],
+                'campaign_start_date': post.campaign_start_date,
+                'campaign_end_date': post.campaign_end_date,
+                'post': post.post,
+                'post_ids': post.ids,
+                post._contact_point_to_vector(): post.apply_vector,
+            })
+            wizard._post_job()
+
+    def _delete_post(self): ...
```

Comment:
Not fond of prototyping in standard codebase.
```suggestion
    def _delete_post(self):
        pass
```